src/design_notes/autoref_ops.md

```diff
@@ -0,0 +1,29 @@
+# Autoref/Autoderef in operators
+
+Rust permits overriding most of the operators (e.g., `+`, `-`, `+=`). In part
+due to `Copy` types not auto-dereferencing, it is common to have `T + &T`
+or `&T + &T`, with potentially many levels of indirection on either side of the
+operator.
+
+There is desire in general to avoid needing to both add impls for referenced
+versions of types because they:
+
+- bloat documentation
+- are never quite sufficient (always more references are possible)
+- can cause inference regressions, as the compiler cannot in general know that
+  `&T + &T` is equivalent to `u64 + u64`.
+
```

Comment:
```suggestion

However, generic code may want the impls on references because the generic code may not want to require `T: Copy`.  Another version of this could involve something like `default impl<T:Copy> Add<&T> for &T` so that people don't *need* to write them themselves.

```